io/plugins/eu.esdihumboldt.hale.io.gml/src/eu/esdihumboldt/hale/io/gml/reader/internal/wfs/DuplicateIDsFilterIterator.java

```diff
@@ -0,0 +1,159 @@
+
+	private final HashSet<String> uniqueIDInstances = new HashSet<String>();
+	private final InstanceIterator iterator;
+	private Instance nextInstance;
+	private boolean instranceAlreadyReturned = false;
+
+	/**
+	 * @param instanceIterator InstanceIterator
+	 */
+	public DuplicateIDsFilterIterator(InstanceIterator instanceIterator) {
+		this.iterator = instanceIterator;
+	}
+
+	/**
+	 * 
+	 * @see eu.esdihumboldt.hale.common.instance.model.ResourceIterator#close()
+	 */
+	@Override
+	public void close() {
+		iterator.close();
+		nextInstance = null;
+		instranceAlreadyReturned = true;
+	}
+
+	/**
+	 * @return TypeDefinition
+	 * @see eu.esdihumboldt.hale.common.instance.model.ext.InstanceIterator#typePeek()
+	 */
+	@Override
+	public TypeDefinition typePeek() {
+		hasNext();
+		if (nextInstance == null) {
+			return null;
+		}
+		return nextInstance.getDefinition();
+	}
+
+	/**
+	 * @return boolean
+	 * @see eu.esdihumboldt.hale.common.instance.model.ext.InstanceIterator#supportsTypePeek()
+	 */
+	@Override
+	public boolean supportsTypePeek() {
+		return true;
+	}
+
+	/**
+	 * 
+	 * @see eu.esdihumboldt.hale.common.instance.model.ext.InstanceIterator#skip()
+	 */
+	@Override
+	public void skip() {
+		next();
+	}
+
+	/**
+	 * @return boolean
+	 * @see java.util.Iterator#hasNext()
+	 */
+	@Override
+	public boolean hasNext() {
+		if (nextInstance == null || instranceAlreadyReturned) {
+
+			while (true) {
+				if (iterator != null && !iterator.hasNext()) {
+					return false;
+				}
+				else {
+					Instance instance = iterator.next();
+					if (!isTheInstancePresent(instance)) {
+						nextInstance = instance;
+						instranceAlreadyReturned = false;
+						return true;
+					}
+				}
+			}
+		}
+		else {
+			return true;
+		}
+	}
+
+	/**
+	 * @return Instance
+	 * @see java.util.Iterator#next()
+	 */
+	@Override
+	public Instance next() {
+		hasNext();
+		if (nextInstance == null) {
+			for (String string : uniqueIDInstances) {
+				System.out.println(string);
+			}
+			throw new NoSuchElementException();
+		}
+		instranceAlreadyReturned = true;
+		return nextInstance;
+	}
+
+	private boolean isTheInstancePresent(Instance instance) {
+		for (QName propertyName : instance.getPropertyNames()) {
+			if (isGmlIdProperty(propertyName)) {
+				Object[] gmlID = instance.getProperty(propertyName);
+				if (gmlID[0] != null) {
+					String gmlIDToCheck = (String) gmlID[0];
+
+					if (!uniqueIDInstances.contains(gmlIDToCheck)) {
+						uniqueIDInstances.add(gmlIDToCheck);
+						return false;
+					}
+					else {
+						return true;
+					}
+
+				}
+			}
+		}
+		return true;
+	}
+
+	private boolean isGmlIdProperty(QName propertyName) {
+		return (propertyName.getNamespaceURI().startsWith(GMLConstants.NS_WFS)
+				|| propertyName.getNamespaceURI().startsWith(GMLConstants.GML_NAMESPACE_CORE))
+				&& "id".equals(propertyName.getLocalPart())
+				&& "gml".equals(propertyName.getPrefix());
```

Comment:
I would leave out checking the prefix, since it is not guaranteed to be `gml`. Checking the namespace is enough.